src/h5web/vis-packs/core/heatmap/config.tsx

```diff
@@ -0,0 +1,105 @@
+import create, {
+  EqualityChecker,
+  State,
+  StateSelector,
+  UseStore,
+} from 'zustand';
+import { persist } from 'zustand/middleware';
+import type { ColorMap } from './models';
+import { CustomDomain, Domain, ScaleType } from '../models';
+import { createContext, useContext, useState } from 'react';
+import { assertDefined } from '../../../guards';
+import type { ConfigProviderProps } from '../../models';
+
+interface HeatmapConfig extends State {
+  dataDomain: Domain | undefined;
+  setDataDomain: (dataDomain: Domain) => void;
+
+  customDomain: CustomDomain;
+  setCustomDomain: (customDomain: CustomDomain) => void;
+
+  colorMap: ColorMap;
+  setColorMap: (colorMap: ColorMap) => void;
+
+  invertColorMap: boolean;
+  toggleColorMapInversion: () => void;
+
+  scaleType: ScaleType;
+  setScaleType: (scaleType: ScaleType) => void;
+
+  keepAspectRatio: boolean;
+  toggleAspectRatio: () => void;
+
+  showGrid: boolean;
+  toggleGrid: () => void;
+}
+
+function initialiseStore() {
```

Comment:
The diff is crap because I renamed `config.ts` to `config.tsx`, but here I just wrap the creation in a function so it can be called lazily by `useState` (see below).